File: 프로그래머스/2/42860. 조이스틱/조이스틱.js
function solution(name) {
    let answer = 0;
    
    let cnt = 0;
    let move = name.length - 1;
    
    for(let i = 0; i < name.length; i++) {
        answer += Math.min(name[i].charCodeAt() - 'A'.charCodeAt(), + 'Z'.charCodeAt() - name[i].charCodeAt() + 1);
        
        let idx = i + 1;
        while(idx < name.length) {
            if(name[idx] === 'A')
                idx++;
            else
                break;
        }
        
        cnt = name.length - 1 - (idx - 1);
        move = Math.min(move, i + cnt + Math.min(i, cnt));
    }
    
    return answer + move;
}

File: 프로그래머스/2/42860. 조이스틱/조이스틱.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(string name) {
    int answer = 0;
    
    int cnt = 0;
    int move = name.length() - 1;
    for(int i = 0; i < name.length(); i++) {
        // 위아래
        answer += min(name[i] - 'A', 'Z' - name[i] + 1);        
        
        int idx = i + 1;
        while(idx < name.length()) {
            if(name[idx] == 'A')
                idx++;
            else
                break;
        }
        
        cnt = name.length() - 1 - idx + 1;
        move = min(move, i + cnt + min(i, cnt));         
    }
    
    return answer + move;
}